3party/CMakeLists.txt

```diff
@@ -5,6 +5,15 @@ set(CMAKE_WARN_DEPRECATED OFF CACHE BOOL "" FORCE)
 if (WITH_SYSTEM_PROVIDED_3PARTY)
   set(GFLAGS_USE_TARGET_NAMESPACE ON)
   find_package(gflags REQUIRED GLOBAL)
+
+  set(_backup_CMAKE_FIND_PACKAGE_TARGETS_GLOBAL ${CMAKE_FIND_PACKAGE_TARGETS_GLOBAL})
+  set(CMAKE_FIND_PACKAGE_TARGETS_GLOBAL ON)
```

Comment:
`find_package(... GLOBAL)` is also CMake 3.24 feature. Maybe it's time to bump CMake version?

Reply:
Right, I see... No, we can't increase it yet due to lower Android cmake version.

What is the reason for promoting packages to a global scope? What are other possible solutions here?

Reply:
> we can't increase it yet due to lower Android cmake version

Why can't it be updated? I mean, isn't cmake used to build android version is the same as the one used to build desktop version?

Reply:
> What is the reason for promoting packages to a global scope? What are other possible solutions here?

find_package results are not visible outside of current directory. Possible solution: move package searching to the root directory.

Reply:
> Why can't it be updated? I mean, isn't cmake used to build android version is the same as the one used to build desktop version?

No, it is shipped with Android SDK.

<img width="730" alt="выява" src="https://github.com/organicmaps/organicmaps/assets/170263/76a3832c-6734-4623-a4ef-b4e92f6f8d89">

Let's consider a different solution then. It would be easier to merge C++ fixes separately from cmake fixes that may be more complex.

Reply:
> No, it is shipped with Android SDK.

It's shipped for convenience; you can use any cmake you want https://developer.android.com/studio/projects/install-ndk#vanilla_cmake

> It would be easier to merge C++ fixes separately from cmake fixes that may be more complex.

Ok, I'll move C++ changes into a separate PR.

Reply:
> you can use any cmake you want

That's not easy. Google heavily patched their cmake in the past to work properly with android. Forcing all android devs to setup a custom sideloaded cmake and crossing fingers for it to work is not an option.

We may consider cmake upgrade later when Android SDK officially updates it.

Reply:
Hmmm https://developer.android.com/ndk/guides/cmake
> Warning: CMake has its [own built-in NDK support](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-android). Before CMake 3.21, this workflow is not supported by Android and is often broken with new NDK releases. Starting from CMake 3.21, the implementations are merged. CMake's built-in support has similar behavior to the NDK toolchain file, though variable names differ. Starting from Android NDK r23, the toolchain file will delegate to CMake's built-in support when using CMake 3.21 or later. See [Issue 463](https://github.com/android/ndk/issues/463) for more information. Note that the Android Gradle Plugin still uses the NDK's toolchain file automatically.

Reply:
How does it help to automatically provide the correct CMake version/binary for Android Studio users?